packages/pluggableWidgets/progress-bar-web/src/components/ProgressBar.tsx

```diff
@@ -0,0 +1,73 @@
+import { createElement, CSSProperties, ReactElement, ReactNode } from "react";
+import classNames from "classnames";
+import { Alert } from "@widgets-resources/piw-utils";
+
+export interface ProgressBarProps {
+    class: string;
+    style?: CSSProperties;
+    currentValue: number;
+    minValue: number;
+    maxValue: number;
+    onClick: (() => void) | undefined;
+    label: ReactNode;
+}
+
+function calculatePercentage(currentValue: number, minValue: number, maxValue: number): number {
+    if (currentValue < minValue) {
+        return 0;
+    }
+    if (currentValue > maxValue) {
+        return 100;
+    }
+    const range = maxValue - minValue;
+    const percentage = Math.round(((currentValue - minValue) / range) * 100);
+    return Math.abs(percentage);
+}
+
+function getValuesErrorMessage(currentValue: number, minValue: number, maxValue: number): string | null {
+    if (maxValue < minValue) {
+        return "Error in progress bar values: The maximum value is lower than the minimum value.";
+    }
+    if (currentValue < minValue) {
+        return "Error in progress bar values: The progress value is lower than the minimum value.";
+    }
+    if (currentValue > maxValue) {
+        return "Error in progress bar values: The progress value is higher than the maximum value.";
+    }
+    return null;
+}
+
+export function ProgressBar({
+    class: className,
+    style,
+    minValue,
+    maxValue,
+    onClick,
+    currentValue,
+    label
+}: ProgressBarProps): ReactElement {
+    const errorMessage = getValuesErrorMessage(currentValue, minValue, maxValue);
+    return (
+        <div className={classNames("widget-progress-bar", "progress-bar-medium", className)} style={style}>
+            <div
+                className={classNames("progress", {
+                    "widget-progress-bar-alert": maxValue < 1,
+                    "widget-progress-bar-clickable": !!onClick
+                })}
+                onClick={onClick}
+            >
+                <div
+                    className={classNames("progress-bar", "progress-bar-default", "progress-bar-medium", className)}
```

Comment:
@diego-antonelli also not quite sure whether this is a good way to implement design properties, because basically all design properties classes are now applied to both this element as well as L51. But different design properties do cover different elements, so it's kind of necessary. I sat with Jordan and we thought it was fine since we can use CSS selectors to target one and not the other, but can you check whether this is fine?

Reply:
So I should remove `className` on this line (60) and then change the CSS selectors to target this element through the element on L51?

Reply:
Yeap